tm2/pkg/sdk/auth/ante.go

```diff
@@ -139,11 +139,19 @@ func NewAnteHandler(ak AccountKeeper, bank BankKeeperI, sigGasConsumer Signature
 		stdSigs := tx.GetSignatures()
 
 		for i := 0; i < len(stdSigs); i++ {
+			var isNewAccount bool
 			// skip the fee payer, account is cached and fees were deducted already
 			if i != 0 {
 				signerAccs[i], res = GetSignerAcc(newCtx, ak, signerAddrs[i])
+				// only create new account when tx is a sponsor transaction
 				if !res.IsOK() {
-					return newCtx, res, true
+					if tx.IsSponsorTx() {
+						isNewAccount = true
```

Comment:
The last parameter here should be renamed to `isNewAccount`, not `isGenesis`. https://github.com/gnolang/gno/pull/2630/files#diff-b9019e160b0344a18489c4d74194d8da800f863a3130ab78e5d24e8d7635193cR444

Then you should do the same in this function; making isNewAccount a variable which is true either because it's an unknown account in a sponsor tx, or because it's a genesis tx.

Reply:
e6d642e581d7de69db193d0e92897a8a344a7255